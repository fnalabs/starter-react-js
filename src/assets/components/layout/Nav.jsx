import React from 'react'
import { Link } from 'react-router-dom'

export const Nav = () => (
  <nav>
    <Link to='/'>Home</Link>
  </nav>
)
