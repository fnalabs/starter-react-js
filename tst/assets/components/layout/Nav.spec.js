/* eslint-env jest */
import React from 'react'
import { MemoryRouter } from 'react-router'
import renderer from 'react-test-renderer'

import { Nav } from '../../../../src/assets/components/layout/Nav.jsx'

describe('<Nav />', () => {
  test('should render homepage nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should render Cookie Policy page nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/cookie']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })

  test('should render Privacy Policy page nav correctly', () => {
    const tree = renderer
      .create(<MemoryRouter initialEntries={['/privacy']}><Nav /></MemoryRouter>)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
