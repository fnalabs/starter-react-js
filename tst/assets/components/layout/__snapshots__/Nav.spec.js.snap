// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> should render Cookie Policy page nav correctly 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</nav>
`;

exports[`<Nav /> should render Privacy Policy page nav correctly 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</nav>
`;

exports[`<Nav /> should render homepage nav correctly 1`] = `
<nav>
  <a
    href="/"
    onClick={[Function]}
  >
    Home
  </a>
</nav>
`;
